beacon_node/beacon_chain/src/builder.rs

```diff
@@ -555,6 +556,9 @@ where
                 weak_subj_state.clone(),
             )
             .map_err(|e| format!("Failed to set checkpoint state as finalized state: {:?}", e))?;
+        // TODO(hdiff): This put state fails is most cases because the start slot is set to 0, and
```

Comment:
Should add the initialisation of the `start_slot` to `init_anchor_info`. I think it should be equal to `weak_subj_slot` (slot of checkpoint state, which is on an epoch boundary).